client/src/components/App.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import Listing from './Listing.jsx'
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      data: [],
+      currentListingId: '',
+      isMounted: false
+    }
+  }
+
+  componentDidMount() {
+    fetch(`http://localhost:3000/api/listing`)
+      .then(res => res.json())
+      .then(
+        (result) => {
+          this.setState({
+            data: result,
+            currentListingId: result[Math.floor(Math.random() * 100)]._id,
```

Comment:
Use window.location.pathname instead to get the url passed in from the user. We want to send back the same room data each time a user navigates to /rooms/:id instead of random data.